chromium_src/components/component_updater/component_installer.cc

```diff
@@ -1,8 +1,48 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
- * You can obtain one at http://mozilla.org/MPL/2.0/. */
+ * you can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "components/component_updater/component_installer.h"
+
+#include <algorithm>
 
 #include "components/crx_file/crx_verifier.h"
 #define CRX3_WITH_PUBLISHER_PROOF CRX3
+#define Register Register_ChromiumImpl
 #include "../../../../components/component_updater/component_installer.cc"
+#undef Register
 #undef CRX3_WITH_PUBLISHER_PROOF
+
+namespace component_updater {
+
+void ComponentInstaller::Register(ComponentUpdateService* cus,
+                                  base::OnceClosure callback) {
+  static std::vector<std::string> blacklisted_components({
+      "bklopemakmnopmghhmccadeonafabnal",  // Legacy TLS Deprecation Config
+      "cmahhnpholdijhjokonmfdjbfmklppij",  // Federated Learning of Cohorts
+      "eeigpngbgcognadeebkilcpcaedhellh",  // Autofill States Data
+      "gcmjkmgdlgnkkcocmoeiminaijmmjnii",  // Subresource Filter Rules
+      "ggkkehgbnfjpeggfpleeakpidbkibbmn",  // Crowd Deny
+      "giekcmmlnklenlaomppkphknjmnnpneh",  // Certificate Error Assistant
+      "jflookgnkcckhobaglndicnbbgbonegd",  // Safety Tips
+      "llkgjffcdpffmhiakmfcdcblohccpfmo",  // Origin Trials
+      "ojhpjlocmbogdgmfpkhlaaeamibhnphh",  // Zxcvbn Data Dictionaries
+#if defined(OS_ANDROID)
+      "lmelglejhemejginpboagddgdfbepgmp",  // Optimization Hints
+      "obedbbhbpmojnkanicioggnmelmoomoc"   // OnDeviceHeadSuggest
+#endif
+  });
+  std::vector<uint8_t> hash;
+  installer_policy_->GetHash(&hash);
```

Comment:
The original `Register` method checks for `installer_policy_` not being `null`, might be good to do the same before using it.

Reply:
Great point. I've added this check and also `DCHECK_CALLED_ON_VALID_THREAD` from the original `Register` method in e8671feb.